src/test/java/dev/openfeature/sdk/OpenFeatureClientTest.java

```diff
@@ -78,23 +83,99 @@ void mergeContextTest() {
     @DisplayName("addHooks should allow chaining by returning the same client instance")
     void addHooksShouldAllowChaining() {
         OpenFeatureAPI api = mock(OpenFeatureAPI.class);
-        OpenFeatureClient client = new OpenFeatureClient(api, "name", "version");
+        OpenFeatureClient client = new OpenFeatureClient(() -> null, api, "name", "version");
         Hook<?> hook1 = Mockito.mock(Hook.class);
         Hook<?> hook2 = Mockito.mock(Hook.class);
 
         OpenFeatureClient result = client.addHooks(hook1, hook2);
-        assertEquals(client, result);  
+        assertEquals(client, result);
     }
 
     @Test
     @DisplayName("setEvaluationContext should allow chaining by returning the same client instance")
     void setEvaluationContextShouldAllowChaining() {
         OpenFeatureAPI api = mock(OpenFeatureAPI.class);
-        OpenFeatureClient client = new OpenFeatureClient(api, "name", "version");
+        OpenFeatureClient client = new OpenFeatureClient(() -> null, api, "name", "version");
         EvaluationContext ctx = new ImmutableContext("targeting key", new HashMap<>());
 
         OpenFeatureClient result = client.setEvaluationContext(ctx);
         assertEquals(client, result);
     }
-   
+
+    @Test
+    @DisplayName("Should not call evaluation methods when the provider has state FATAL")
+    void shouldNotCallEvaluationMethodsWhenProviderIsInFatalErrorState() {
+        MockProvider mockProvider = new MockProvider(ProviderState.FATAL);
+        OpenFeatureAPI api = mock(OpenFeatureAPI.class);
+        OpenFeatureClient client = new OpenFeatureClient(() -> mockProvider, api, "name", "version");
+        FlagEvaluationDetails<Boolean> details = client.getBooleanDetails("key", true);
+
+        assertThat(mockProvider.isEvaluationCalled()).isFalse();
+        assertThat(details.getErrorCode()).isEqualTo(ErrorCode.PROVIDER_FATAL);
+    }
+
+    @Test
+    @DisplayName("Should not call evaluation methods when the provider has state NOT_READY")
+    void shouldNotCallEvaluationMethodsWhenProviderIsInNotReadyState() {
+        MockProvider mockProvider = new MockProvider(ProviderState.NOT_READY);
+        OpenFeatureAPI api = mock(OpenFeatureAPI.class);
+        OpenFeatureClient client = new OpenFeatureClient(() -> mockProvider, api, "name", "version");
+        FlagEvaluationDetails<Boolean> details = client.getBooleanDetails("key", true);
+
+        assertThat(mockProvider.isEvaluationCalled()).isFalse();
+        assertThat(details.getErrorCode()).isEqualTo(ErrorCode.PROVIDER_NOT_READY);
+    }
+
+    private static class MockProvider implements FeatureProvider {
```

Comment:
You might be able to use the `TestEventsProvider` for this but that's up to you.